mapit_global/management/commands/mapit_global_update_countries.py

```diff
@@ -0,0 +1,129 @@
+            country_code = tags[0].attrib['v']
+            Code.objects.create(area=area, type=self.iso_code_type, code=country_code)
+
+    def ensure_countries_exist(self):
+        for code_object in self.iso_code_type.codes.select_related('area'):
+            Country.objects.get_or_create(
+                code=code_object.code,
+                defaults={'name': code_object.area.name})
+        Country.objects.get_or_create(
+            code='?',
+            defaults={'name': 'Multiple enclosing countries'})
+
+    @cached_property
+    def iso_code_type(self):
+        return CodeType.objects.get_or_create(
+            code='iso3166_1',
+            defaults={'description': 'ISO3166-1 country code from OSM'})[0]
+
+    def get_enclosing_country_codes(self, area):
+        enclosing_country_codes = set()
+        for polygon_object in area.polygons.all():
+            point = polygon_object.polygon.point_on_surface
+            # Now find the country-level areas that contain that point:
+            for enclosing_country_area in Area.objects.filter(
+                    type__code='O02',
+                    codes__type=self.iso_code_type,
+                    polygons__polygon__contains=point,
+                    **self.generation_kwargs()
+            ):
+                country_code_object = enclosing_country_area.codes.get(type=self.iso_code_type)
+                enclosing_country_codes.add(country_code_object.code)
+        return tuple(sorted(enclosing_country_codes))
+
+    def get_country_from_codes(self, country_codes):
+        if not country_codes:
+            return None
+        if len(country_codes) > 1:
+            unique_country_code = '?'
+        else:
+            unique_country_code = next(iter(country_codes))
+        return self.code_to_country[unique_country_code]
+
+    def set_country_on_all_areas(self):
+        error_messages = []
+        for area in Area.objects.filter(**self.generation_kwargs()):
+            print("Considering area:", area, area.id)
+            enclosing_country_codes = self.get_enclosing_country_codes(area)
+            country = self.get_country_from_codes(enclosing_country_codes)
+            if country and country.code == '?':
+                msg = 'Multiple enclosing country codes {codes} found for area {name} with ID {id}'
+                error_messages.append(msg.format(
+                    codes=unicode(enclosing_country_codes),
+                    name=area.name,
+                    id=area.id))
+            area.country = country
+            area.save()
+        return error_messages
+
+    @cached_property
+    def code_to_country(self):
+        return {country.code: country for country in Country.objects.all()}
+
+    def handle(self, **options):
+        generation_id = options['generation']
+        if not generation_id:
+            raise CommandError('You must specify a generation with --generation')
```

Comment:
Can't you put `required=True` on the `add_argument` call?

Reply:
Good point, thanks